lib/puppet/transaction/report.rb

```diff
@@ -127,6 +127,11 @@ class Puppet::Transaction::Report
   #      corrective changes.
   attr_reader :corrective_change
 
+  # @return [Boolean] true if one or more resources attempted to generate
+  #   resources and failed
+  #
+  attr_accessor :resources_failed_to_generate
```

Comment:
I don't include this in the initializer(s) but I'm wondering if I should, specifically `initialize_from_hash` ?